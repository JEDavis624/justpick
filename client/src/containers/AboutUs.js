import React, { Component } from 'react';
import MainJumbotron from '../components/Jumbotron';
import { Row, Col } from 'react-bootstrap';

class AboutUs extends Component {

    render() {
        return(
            <div>
                <MainJumbotron />

                <Row>
                    <h2 className="aboutUsHeaders">About Us</h2>
                </Row>

                <Row>
                    <Col md={1}></Col>
                    <Col md={10}>
                        <p> JustPick - a web application will allow individuals to form teams, set times and find locations to play.Creating a pick up game, for any sport, is difficult to orchestrate and requires one to personally know a lot of individuals that are interested in the same sport as them. This requires several people to find a mutually agreeable time and location to play games. Some people do not have the time to commit to a league and are only able to play every once in a while.
                            As an athlete, I want to provide other athletes with the ability to easily organize pick up games for any sport, so athletes can connect with other athletes in their area. I also want to give other athletes a chance to feel like they are actually playing for something important and not just playing for fun. </p>
                    </Col>
                        <Col md={1}></Col>
                </Row>

                <Row>
                    <h2 className="aboutUsHeaders">Meet the team</h2>
                </Row>

                <Row>
                    
                    <Col md={4}>
                        <div>
                            <h3 className="aboutUsHeaders"> Jariel Brown </h3>
                            <img id="avatar" src="/images/jariel.jpg" alt="Jariel Brown"></img>
                            <p id="autobio">Jariel is the resident back-end expert of JustPick. If a route needs an endpoint or a database needs a schema - Jariel is your man. He has gone to great lengths to make sure the user experience is seamless from start to finish. Jariel's ability to solve complex problems for JustPick has shown through time and time again.</p>
                        </div>
                    </Col>
                    <Col md={4}>
                        <h3 className="aboutUsHeaders"> Mesut Sipar </h3>
                        <img id="avatar" src="/images/mesut.png" alt="Mesut Sipar"></img>
                        <p id="autobio">Mesut is the ultimate utility man. The next time you successfully book a lesson - make sure to send a kind note Mesut's way - because that lesson would never have been possible without his blood, sweat and tears poured out over the calendar system currently in place for JustPick. And oh by the way - he can make a mean wireframe as well.</p>
                    </Col>
                    <Col md={4}>
                        <h3 className="aboutUsHeaders"> John Davis </h3>
                        <img id="avatar" src="/images/john.jpg" alt="John Davis"></img>
                        <p id="autobio">John Davis is the cleanup man of the crew. He was primarily responsible for the look & feel of ParTee as well as it's overall goals and mission. Don't forget that everytime you upload a mugshot of yourself to your profile - Clarke holds all the records in his Cloudinary database - choose wisely! </p>
                    </Col>
            
                </Row>
            </div>
        )

    }
}

export default AboutUs;