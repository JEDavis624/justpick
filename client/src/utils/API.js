import axios from 'axios';

export default {
    createAccount: function(user) {
        return axios.post('/api/signup', user);
    },

    login: function(user) {
        return axios.post('/api/login', user);
    },

    logout: function() {
        return axios.get('/api/logout');
    },

    update: function(updates) {
        return axios.post('/api/profile', updates);
    },

    // update: function() {
    //     return axios.post('/api/about');
    // },
    
    getAllPros: function() {
        return axios.get('/api/pros')
    },

    getAppointmentsByUserId: function(id) {
        return axios.get(`/api/pros/${id.toString()}/appointments`);
    },

    createAppointment: function(instructor, time) {
        const data = { instructor, time };
        console.log(data);
        return axios.post(`/api/appt`, data);
    },

    getAvailability: function() {
        return axios.get('/api/availability');
    },

    getReviewAuth: function(instructorId) {
        return axios.post('/api/review/auth', { instructorId });
    },

    createReview: function(instructorId, score) {
        return axios.post('/api/review', { instructorId, score });
    }
};

