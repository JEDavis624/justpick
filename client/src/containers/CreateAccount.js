import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { LinkContainer } from 'react-router-bootstrap';
import { Button, FormGroup, ControlLabel, 
         Radio, Row, Col, Panel, Alert } from 'react-bootstrap';
import FieldGroup from '../components/FieldGroup';
import MainJumbotron from '../components/Jumbotron';
import API from '../utils/API';

export class CreateAccount extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            firstName: '',
            lastName: '',
            isPro: false,
            badCreate: false,
            errorMsg: '',
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleAccountCreate = this.handleAccountCreate.bind(this);
    }

    handleInputChange (event) {
        const { name, value } = event.target;
        this.setState({
            [name]: value
        });
    };

    handleAccountCreate (event) {
        event.preventDefault();
        const user = {
            email: this.state.email,
            password: this.state.password,
            firstName: this.state.firstName,
            lastName: this.state.lastName,
            isPro: this.state.isPro
        };
        API.createAccount(user)
        .then(res => {
            if (res.status === 200) {
                debugger;
                this.setState({
                    email: '',
                    password: '',
                    firstName: '',
                    lastName: '',
                    isPro: false
                });
                this.props.setAuth(true, res.data.user);
                this.props.history.push("/pro");
            } else {
                console.log(res.data);
            }
        })
        .catch(err => {
            const errData = err.response.data;
            if (errData.message[0] && errData.message[0] === 'duplicate email'){
                this.setState({ errorMsg: "That email address is already in use" });
            } else {
                this.setState({ errorMsg: "You did something wrong.  Maybe fix that..."});
            }
            this.setState({ badCreate: true })
        });
    }

    render() {
        return (
            <div>

            <MainJumbotron />

                <Row>
                <Col xs={12} sm={8} md={4} smOffset={2} mdOffset={4}>
                    <Panel bsStyle="primary">
                        <Panel.Heading>Create an account</Panel.Heading>
                        <Panel.Body>
                            <form onSubmit={this.handleAccountCreate}>
                                <FieldGroup
                                    name="email"
                                    value={this.state.email}
                                    type="email"
                                    label="Email address"
                                    placeholder="Enter an email address"
                                    onChange={this.handleInputChange}
                                />
                                <FieldGroup
                                    name="password"
                                    value={this.state.password}
                                    type="password"
                                    label="Password"
                                    placeholder="Enter a password"
                                    help="Password must be at least 6 characters."
                                    onChange={this.handleInputChange}
                                />
                                <FieldGroup
                                    name="firstName"
                                    value={this.state.firstName}
                                    type="text"
                                    label="First name"
                                    placeholder="Enter your first name"
                                    onChange={this.handleInputChange}
                                />
                                <FieldGroup
                                    name="lastName"
                                    value={this.state.lastName}
                                    type="text"
                                    label="Last name"
                                    placeholder="Enter your last name"
                                    onChange={this.handleInputChange}
                                />
                                <FormGroup>
                                    <ControlLabel>Who are you?</ControlLabel>
                                    <Radio name="isPro" value="false" defaultChecked onChange={this.handleInputChange}>
                                        I'm here to learn.
                                    </Radio>
                                    <Radio name="isPro" value="true" onChange={this.handleInputChange}>
                                        I'm pro!
                                    </Radio>
                                </FormGroup>
                                {
                                    this.state.badCreate &&
                                    <Alert bsStyle="danger">
                                        <p>{this.state.errorMsg}</p>
                                    </Alert>
                                }
                                <Button 
                                    bsStyle="primary" 
                                    type="submit"
                                    block
                                >
                                    Create Account
                                </Button>
                            </form>
                        </Panel.Body>
                        <Panel.Footer style={{textAlign: 'center'}}>
                            <LinkContainer to={"/login"}><a>I already have an account. I'd like to log in.</a></LinkContainer>
                        </Panel.Footer>
                    </Panel>
                </Col>
            </Row>
            </div>
        );
    }
}

export default withRouter(CreateAccount);