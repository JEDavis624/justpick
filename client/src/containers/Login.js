import React, { Component } from 'react';
import { Button, Panel, Alert } from 'react-bootstrap';
import FieldGroup from '../components/FieldGroup';
import { withRouter } from 'react-router-dom';
import { LinkContainer } from 'react-router-bootstrap';
import MainJumbotron from '../components/Jumbotron';
import API from '../utils/API';

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            badLogin: false
        };
    }

    handleInputChange = event => {
        const { name, value } = event.target;
        this.setState({
            [name]: value
        });
    };

    handleLogin = event => {
        event.preventDefault();
        console.log(`Logging in ${this.state.email} with password ${this.state.password}`);
        const { email, password } = this.state;
        console.log(email, password);
        API.login({ email, password })
            .then((res) => {
                console.log(res);
                if (res.status === 200) {
                    this.props.setAuth(true, res.data);
                    this.props.history.push("/");
                }
            })
            .catch((e) => {
                console.log(e);
                this.setState({ badLogin: true })
            });
    };

    render() {
        return (
            <div>
            <MainJumbotron />
            <Row>
                <Col xs={12} sm={8} md={4} smOffset={2} mdOffset={4}>
                    <Panel bsStyle="primary">
                        <Panel.Heading>Log in</Panel.Heading>
                        <Panel.Body>
                            <form>
                                <FieldGroup
                                    name="email"
                                    type="email"
                                    label="Email address"
                                    placeholder="Enter your email address"
                                    onChange={this.handleInputChange}
                                />
                                <FieldGroup
                                    name="password"
                                    type="password"
                                    label="Password"
                                    placeholder="Enter your password"
                                    onChange={this.handleInputChange}
                                />
                                {
                                    this.state.badLogin &&
                                    <Alert bsStyle="danger">
                                        <p>Invalid Credentials</p>
                                    </Alert>
                                }
                                <Button 
                                    type="submit" 
                                    bsStyle="primary" 
                                    onClick={this.handleLogin}
                                    block
                                >
                                    Log in
                                </Button>
                            </form>
                        </Panel.Body>
                        <Panel.Footer style={{textAlign: 'center'}}>
                            <LinkContainer to={"/createaccount"}><a>Don't have an account? Create one!</a></LinkContainer>
                        </Panel.Footer>
                    </Panel>
                </Col>
            </Row>
            </div>
        );
    }
}

export default withRouter(Login);