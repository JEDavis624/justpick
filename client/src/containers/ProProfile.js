import React, { Component } from 'react';
import { FormGroup, Radio, Checkbox, Button, FormControl, ControlLabel, Alert } from 'react-bootstrap';
import { Link } from 'react-router-dom';
// import FieldGroup from '../components/FieldGroup';
import BigCalendar from 'react-big-calendar';
import moment from 'moment';
import API from '../utils/API';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import CloudProfilePic from '../components/CloudProfilePic';
import { render } from 'react-dom';


BigCalendar.momentLocalizer(moment);

class ProProfile extends Component {
    constructor(props) {
        super(props);

        const miniTime = new Date();
        miniTime.setHours(8,0,0);
        const maxiTime = new Date();
        maxiTime.setHours(20,0,0);
        const defaultDate = new Date();
        defaultDate.setHours(0, 0, 0);

        this.state = {
            junior: false,
            beginner: false,
            intermediate: false,
            advanced: false,

            driving: false,
            approach: false,
            shortGame: false,
            putting: false,

            zipCode: "",
            name: "",
            bio: "",
            photo: "",
            isPro: false,
            rate: "",

            updated: false,
            radioChecked: "",

            events: [],
            minTime: miniTime,
            maxTime: maxiTime,
            defaultDate: defaultDate
        }
    }

    componentWillMount() {
        this.setState({isPro: this.props.user.isPro});
    }

    onUpdateProfile = () => {
        const skills = ["junior", "beginner", "intermediate", "advanced"];
        let skillLevel = [];

        const areas = ["driving", "approach", "shortGame", "putting"];
        let instruction = [];
        for (const skill in skills) {
            if (this.state[skills[skill]]) { skillLevel.push(skills[skill]) }
        }
        console.log(skillLevel)

        for (const area in areas) {
            if (this.state[areas[area]]) { instruction.push(areas[area]) }
        }
        console.log(instruction);

        const updates = {
            bio: this.state.bio,
            photo: this.state.photo,
            zipCode: this.state.zipCode,
            rate: this.state.rate,
            skillLevel,
            instruction
        }
        console.log(updates);
        if (skillLevel.length === 0 || instruction.length === 0){
            this.setState({
                errorMsg: "You must select a Skill Level and Area of Instruction",
                badSubmit: true
            });
        } else {
            API.update(updates).then((res) => {
                console.log(res);
                this.props.updateUser(res.data);
                this.setState({ updated: false })
            }).catch((e) => {
                console.log(e.response.data.errors);
            })
        }

    }

    onCheckboxChange = (e) => {
        const { name, checked } = e.target;
        this.setState({ [name]: checked })
        console.log(name, checked);

        if (!this.state.updated) { this.setState({ updated: true })}
    }

    onRadioChange = (e) => {
        const { value, checked } = e.target;
        console.log(value, checked)
        if (this.state.radioChecked) {
            this.setState({ [this.state.radioChecked]: false });
        }
        this.setState({ 
            [value]: checked,
            radioChecked: value
        });

        if (!this.state.updated) { this.setState({ updated: true })}
    }

    onZipChange = (e) => {
        const { name, value } = e.target;
        if (value.match(/^\d{0,5}$/)){
            this.setState({ [name]: value });
        }
        console.log(name, value);

        if (!this.state.updated) { this.setState({ updated: true })}
    }

    onTextAreaChange = (e) => {
        const { name, value } = e.target;
        this.setState({ [name]: value });
        console.log(name, value);

        if (!this.state.updated) { this.setState({ updated: true })}
    }

    onRateChange = (e) => {
        const { name, value } = e.target;
        if (value.match(/^\d*$/)){
            this.setState({ [name]: value});
        }

        if (!this.state.updated) { this.setState({ updated: true })}
    }

    onPhotoUpdate = (url) => {
        this.setState({ 
            photo: url,
            updated: true
        })
    }

    componentDidMount() {
        console.log("didmount called")
        if (this.props.user && this.props.user.profile) {
            const newBio = this.props.user.profile.bio;
            this.setState({ bio: newBio });

            const newZip = this.props.user.profile.zipCode;
            console.log(newZip);
            this.setState({ zipCode: newZip });

            const skillLevel = this.props.user.profile.skillLevel;
            console.log(skillLevel);
            skillLevel.forEach(skill => this.setState({ 
                [skill]: true,
                radioChecked: skill
            }));

            const instruction = this.props.user.profile.instruction;
            instruction.forEach(area => this.setState({ [area]: true }));

            const photo = this.props.user.profile.photo;
            this.setState({ photo });

            const rate = this.props.user.profile.rate;
            this.setState({ rate });
        }
        const name = `${this.props.user.firstName} ${this.props.user.lastName}`;
        this.setState({ name });
        this.setState({ isPro: this.props.user.isPro });

        API.getAppointmentsByUserId(this.props.user.userId)
            .then(res => {
                // console.log(res.data);
                const events = this.transformApptToEvent(res.data);
                // console.log('Did the transform work?');
                // console.log(events);
                this.setState({
                    events: events
                });
            })
            .catch(err => console.log(err));
        
        //this.setState({ zipCode: newZip })
    }

    transformApptToEvent = (appts) => {
      const events = appts.map(appt => {
          const event = {
              id: appt._id,
              title: 'Sports Lesson',
              start: new Date(appt.startTime),
              end: new Date(appt.startTime + 3600000)
          };
          return event;
      });
      return events;  
    };

    buttonStyle = {
        margin: '20px 0'
    }

    render() {
        return (
            <Grid>
                <Row>
                    <Col xs={6} md={4} className="nameImage">
                        
                         {this.state.photo === '' ? <CloudProfilePic onPhotoUpdate={this.onPhotoUpdate} /> : <Image src={this.state.photo} thumbnail />}
                       
                        <h3 
                        className="profileName"
                        >{this.state.name}</h3>

                    </Col>

                    <Col xs={6} md={8}>
                        <FormGroup>
                            <ControlLabel>Bio</ControlLabel>
                            <FormControl 
                                componentClass="textarea"
                                rows="8"
                                name="bio"
                                value={this.state.bio}
                                placeholder="Tell us about yourself."
                                onChange={this.onTextAreaChange}
                            />
                        </FormGroup>
                    </Col>
                </Row>

                <Row>
                    <Col xs={6} md={4}>
                        {
                            this.props.user.isPro && this.props.user.profile === null &&
                            <Alert bsStyle="warning">
                                <h4>Attention</h4>
                                <p>You must create a profile to setup your availability, and you must setup your availability to show up in search results</p>
                            </Alert>
                        }
                    </Col>

                    <Col xs={6} md={4}>
                        <FormGroup>
                            <ControlLabel>Zip Code</ControlLabel>
                            <FormControl
                                name="zipCode"
                                value={this.state.zipCode}
                                placeholder="Enter Zip Code"
                                onChange={this.onZipChange}
                            />
                            {/* <input 
                                placeholder="Enter Zip Code"
                                value={this.state.zipCode}
                                name="zipCode"
                                onChange={this.onTextChange}    
                            ></input> */}
                        </FormGroup>
                    </Col>
                </Row>

                <Row>
                    <Col xs={6} md={4}>
                    </Col>

                    <Col xs={6} md={4}>
                        { this.state.isPro ?
                        ( <div><h4>Skill Levels Taught:</h4>
                        <FormGroup>
                            <Checkbox 
                                name="junior"
                                checked={this.state.junior}
                                onChange={this.onCheckboxChange}
                            >Junior</Checkbox>
                            <Checkbox
                                name="beginner"
                                checked={this.state.beginner}
                                onChange={this.onCheckboxChange}
                            >Beginner</Checkbox>
                            <Checkbox
                                name="intermediate"
                                checked={this.state.intermediate}
                                onChange={this.onCheckboxChange}
                            >Intermediate</Checkbox>
                            <Checkbox
                                name="advanced"
                                checked={this.state.advanced}
                                onChange={this.onCheckboxChange}
                            >Advanced</Checkbox>
                        </FormGroup>
                        <ControlLabel>Hourly Rate</ControlLabel>
                        <FormControl
                                name="rate"
                                value={this.state.rate}
                                placeholder="Enter Hourly Rate"
                                onChange={this.onRateChange}
                            />
                        <Link to="/avail"><Button style={this.buttonStyle} bsStyle="success" disabled={this.props.user.profile === null ? true : false}>Set Availability</Button></Link>
                        </div>)

                        : ( <div><h4>Skill Level:</h4>
                        <FormGroup>
                            <Radio
                                name="skillLevel"
                                value="junior"
                                checked={this.state.radioChecked === 'junior'}
                                onChange={this.onRadioChange}
                            >Junior</Radio>
                            <Radio
                                name="skillLevel"
                                value="beginner"
                                checked={this.state.radioChecked === 'beginner'}
                                onChange={this.onRadioChange}
                            >Beginner</Radio>
                            <Radio
                                name="skillLevel"
                                value="intermediate"
                                checked={this.state.radioChecked === 'intermediate'}
                                onChange={this.onRadioChange}
                            >Intermediate</Radio>
                            <Radio
                                name="skillLevel"
                                value="advanced"
                                checked={this.state.radioChecked === 'advanced'}
                                onChange={this.onRadioChange}
                            >Advanced</Radio>
                        </FormGroup></div> )}
                    </Col>

                    <Col xs={6} md={4}>
                        {this.state.isPro ? <h4>Provide Instruction In:</h4> : <h4>Seeking Instruction For:</h4>}
                        <FormGroup>
                            <Checkbox
                                name="football"
                                checked={this.state.driving}
                                onChange={this.onCheckboxChange}
                            >Football</Checkbox>
                            <Checkbox
                                name="soccer"
                                checked={this.state.approach}
                                onChange={this.onCheckboxChange}
                            >Soccer</Checkbox>
                            <Checkbox
                                name="basketball"
                                checked={this.state.shortGame}
                                onChange={this.onCheckboxChange}
                            >Basketball</Checkbox>
                            <Checkbox
                                name="baseball"
                                checked={this.state.putting}
                                onChange={this.onCheckboxChange}
                            >Baseball</Checkbox>
                        </FormGroup>
                    </Col>
                </Row>

                <Row>
                    {
                        this.state.badSubmit &&
                        <Alert bsStyle="danger">
                            <p>{this.state.errorMsg}</p>
                        </Alert>
                    }
                    <Button 
                        bsStyle="primary" 
                        id="updateProfile" 
                        disabled={!this.state.updated}
                        onClick={this.onUpdateProfile}    
                    >Update Profile</Button>
                </Row>
                <Row style={{ height: 800 }} >
                    <BigCalendar 
                        events={this.state.events}
                        defaultDate={this.state.defaultDate}
                        defaultView={this.state.isPro ? 'week' : 'agenda'}
                        step={30}
                        min={this.state.minTime}
                        max={this.state.maxTime}
                        formats={{
                            dayFormat: 'ddd M/D',
                            weekdayFormat: 'dddd'
                        }}
                    />
                </Row>
            </Grid>
        );
    }
}

export default ProProfile;