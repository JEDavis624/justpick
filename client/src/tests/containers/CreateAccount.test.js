import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from 'react-bootstrap';

import { CreateAccount } from '../../containers/CreateAccount';
import API from '../../utils/API';

let wrapper, setAuth, history, handleInputChange, handleAccountCreate, event;
jest.mock('../../utils/API');

const sleep = (ms) => {
  return new Promise(resolve => setTimeout(resolve, ms));
}

describe('CreateAccount', () => {
  beforeEach(() => {
    setAuth = jest.fn();
    history = { push: jest.fn() }
    handleInputChange = jest.spyOn(CreateAccount.prototype, 'handleInputChange');
    handleAccountCreate = jest.spyOn(CreateAccount.prototype, 'handleAccountCreate');
    event = {
      preventDefault: jest.fn()
    }
    wrapper = shallow(<CreateAccount setAuth={setAuth} history={history} />);
  });

  test('Initial render', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('isPro')).toBe(false);
  });

  test('Email input updates state', () => {
    expect(wrapper.state('email')).toEqual('');
    wrapper.find({name: 'email'}).simulate('change', {
      target: {
        name: 'email',
        value: 'john@mail.com'
      }
    });
    expect(handleInputChange).toBeCalled();
    expect(wrapper.state('email')).toEqual('john@mail.com');
  });

  test('Password input updates state', () => {
    expect(wrapper.state('password')).toEqual('');
    wrapper.find({name: 'password'}).simulate('change', {
      target: {
        name: 'password',
        value: 'password'
      }
    });

    expect(handleInputChange).toBeCalled();
    expect(wrapper.state('password')).toEqual('password');
  });

  test('First Name input updates state', () => {
    expect(wrapper.state('firstName')).toEqual('');
    wrapper.find({name: 'firstName'}).simulate('change', {
      target: {
        name: 'firstName',
        value: 'John'
      }
    });

    expect(handleInputChange).toBeCalled();
    expect(wrapper.state('firstName')).toEqual('John');
  });

  test('Last Name input updates state', () => {
    expect(wrapper.state('lastName')).toEqual('');
    wrapper.find({name: 'lastName'}).simulate('change', {
      target: {
        name: 'lastName',
        value: 'Smith'
      }
    });

    expect(handleInputChange).toBeCalled();
    expect(wrapper.state('lastName')).toEqual('Smith');
  });

  test('Pro flag is updated', () => {
    expect(wrapper.state('isPro')).toBe(false);
    wrapper.find({value: 'true'}).simulate('change', {
      target: {
        name: 'isPro',
        value: true
      }
    });

    expect(handleInputChange).toBeCalled();
    expect(wrapper.state('isPro')).toEqual(true);
  });

  test('Duplicate email error is displayed', async () => {
    const rejectedValue = {
        response: {data: { message: ['duplicate email']}}
        
      }
    
    API.createAccount.mockRejectedValue(rejectedValue);
    wrapper.instance().handleAccountCreate(event)
    await sleep(500);
    expect(wrapper.state('badCreate')).toBe(true);
    expect(wrapper.state('errorMsg')).toBe('That email address is already in use');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  test('Other errors displayed', async () => {
    const rejectedValue = {
      response: { data: { message: "error" }}
    }
    API.createAccount.mockRejectedValue(rejectedValue);
    wrapper.instance().handleAccountCreate(event);
    await sleep(500);
    expect(wrapper.state('badCreate')).toBe(true);
    expect(wrapper.state('errorMsg')).toBe('You did something wrong.  Maybe fix that...');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  test('Successful account creation', async () => {
    const resolvedValue = {
      status: 200,
      data: { user: "someone "}
    }
    wrapper.setState({
      email: 'mail@mail.com',
      password: 'password',
      firstName: 'Jane',
      lastName: 'Smith',
      isPro: true
    });

    API.createAccount.mockResolvedValue(resolvedValue);
    wrapper.instance().handleAccountCreate(event);
    await sleep(500);
    expect(setAuth).toBeCalledWith(true, resolvedValue.data.user);
    expect(history.push).toBeCalledWith('/pro');
    expect(wrapper.state()).toEqual({
      email: '',
      password: '',
      firstName: '',
      lastName: '',
      isPro: false,
      badCreate: false,
      errorMsg: ''
    });
  });
});