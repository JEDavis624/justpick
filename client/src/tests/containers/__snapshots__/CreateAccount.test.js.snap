// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount Duplicate email error is displayed 1`] = `
<div>
  <MainJumbotron />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      mdOffset={4}
      sm={8}
      smOffset={2}
      xs={12}
    >
      <Uncontrolled(Panel)
        bsStyle="primary"
      >
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          Create an account
        </PanelHeading>
        <PanelBody
          bsClass="panel"
          collapsible={false}
        >
          <form
            onSubmit={[Function]}
          >
            <FieldGroup
              label="Email address"
              name="email"
              onChange={[Function]}
              placeholder="Enter an email address"
              type="email"
              value=""
            />
            <FieldGroup
              help="Password must be at least 6 characters."
              label="Password"
              name="password"
              onChange={[Function]}
              placeholder="Enter a password"
              type="password"
              value=""
            />
            <FieldGroup
              label="First name"
              name="firstName"
              onChange={[Function]}
              placeholder="Enter your first name"
              type="text"
              value=""
            />
            <FieldGroup
              label="Last name"
              name="lastName"
              onChange={[Function]}
              placeholder="Enter your last name"
              type="text"
              value=""
            />
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Who are you?
              </ControlLabel>
              <Radio
                bsClass="radio"
                defaultChecked={true}
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="false"
              >
                I'm a student. I'm here to learn.
              </Radio>
              <Radio
                bsClass="radio"
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="true"
              >
                I'm a pro. I'm here to teach.
              </Radio>
            </FormGroup>
            <Alert
              bsClass="alert"
              bsStyle="danger"
              closeLabel="Close alert"
            >
              <p>
                That email address is already in use
              </p>
            </Alert>
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              type="submit"
            >
              Create Account
            </Button>
          </form>
        </PanelBody>
        <PanelFooter
          bsClass="panel"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <LinkContainer
            activeClassName="active"
            exact={false}
            replace={false}
            strict={false}
            to="/login"
          >
            <a>
              I already have an account. I'd like to log in.
            </a>
          </LinkContainer>
        </PanelFooter>
      </Uncontrolled(Panel)>
    </Col>
  </Row>
</div>
`;

exports[`CreateAccount Initial render 1`] = `
<div>
  <MainJumbotron />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      mdOffset={4}
      sm={8}
      smOffset={2}
      xs={12}
    >
      <Uncontrolled(Panel)
        bsStyle="primary"
      >
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          Create an account
        </PanelHeading>
        <PanelBody
          bsClass="panel"
          collapsible={false}
        >
          <form
            onSubmit={[Function]}
          >
            <FieldGroup
              label="Email address"
              name="email"
              onChange={[Function]}
              placeholder="Enter an email address"
              type="email"
              value=""
            />
            <FieldGroup
              help="Password must be at least 6 characters."
              label="Password"
              name="password"
              onChange={[Function]}
              placeholder="Enter a password"
              type="password"
              value=""
            />
            <FieldGroup
              label="First name"
              name="firstName"
              onChange={[Function]}
              placeholder="Enter your first name"
              type="text"
              value=""
            />
            <FieldGroup
              label="Last name"
              name="lastName"
              onChange={[Function]}
              placeholder="Enter your last name"
              type="text"
              value=""
            />
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Who are you?
              </ControlLabel>
              <Radio
                bsClass="radio"
                defaultChecked={true}
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="false"
              >
                I'm a student. I'm here to learn.
              </Radio>
              <Radio
                bsClass="radio"
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="true"
              >
                I'm a pro. I'm here to teach.
              </Radio>
            </FormGroup>
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              type="submit"
            >
              Create Account
            </Button>
          </form>
        </PanelBody>
        <PanelFooter
          bsClass="panel"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <LinkContainer
            activeClassName="active"
            exact={false}
            replace={false}
            strict={false}
            to="/login"
          >
            <a>
              I already have an account. I'd like to log in.
            </a>
          </LinkContainer>
        </PanelFooter>
      </Uncontrolled(Panel)>
    </Col>
  </Row>
</div>
`;

exports[`CreateAccount Other errors displayed 1`] = `
<div>
  <MainJumbotron />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={4}
      mdOffset={4}
      sm={8}
      smOffset={2}
      xs={12}
    >
      <Uncontrolled(Panel)
        bsStyle="primary"
      >
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          Create an account
        </PanelHeading>
        <PanelBody
          bsClass="panel"
          collapsible={false}
        >
          <form
            onSubmit={[Function]}
          >
            <FieldGroup
              label="Email address"
              name="email"
              onChange={[Function]}
              placeholder="Enter an email address"
              type="email"
              value=""
            />
            <FieldGroup
              help="Password must be at least 6 characters."
              label="Password"
              name="password"
              onChange={[Function]}
              placeholder="Enter a password"
              type="password"
              value=""
            />
            <FieldGroup
              label="First name"
              name="firstName"
              onChange={[Function]}
              placeholder="Enter your first name"
              type="text"
              value=""
            />
            <FieldGroup
              label="Last name"
              name="lastName"
              onChange={[Function]}
              placeholder="Enter your last name"
              type="text"
              value=""
            />
            <FormGroup
              bsClass="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Who are you?
              </ControlLabel>
              <Radio
                bsClass="radio"
                defaultChecked={true}
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="false"
              >
                I'm a student. I'm here to learn.
              </Radio>
              <Radio
                bsClass="radio"
                disabled={false}
                inline={false}
                name="isPro"
                onChange={[Function]}
                title=""
                value="true"
              >
                I'm a pro. I'm here to teach.
              </Radio>
            </FormGroup>
            <Alert
              bsClass="alert"
              bsStyle="danger"
              closeLabel="Close alert"
            >
              <p>
                You did something wrong.  Maybe fix that...
              </p>
            </Alert>
            <Button
              active={false}
              block={true}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              type="submit"
            >
              Create Account
            </Button>
          </form>
        </PanelBody>
        <PanelFooter
          bsClass="panel"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <LinkContainer
            activeClassName="active"
            exact={false}
            replace={false}
            strict={false}
            to="/login"
          >
            <a>
              I already have an account. I'd like to log in.
            </a>
          </LinkContainer>
        </PanelFooter>
      </Uncontrolled(Panel)>
    </Col>
  </Row>
</div>
`;
